src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -137,6 +138,30 @@ public Task BadRequestIfHostHeaderDoesNotMatchRequestTarget(string requestTarget
                 CoreStrings.FormatBadRequest_InvalidHostHeader_Detail(host.Trim()));
         }
 
+        [Fact]
+        public async Task BadRequestIfHostHeaderDoesNotMatchRequestTarget_OptOut()
```

Comment:
Nit: Does this name need to be changed since it's not supposed to be treated as a bad request in this case?

Reply:
```suggestion
        public async Task CanOptOutOfBadRequestIfHostHeaderDoesNotMatchRequestTarget()
```